### DISCUSSION TITLE
How to use the rest backend?

### DISCUSSION THREAD

**Discussion by SchoolGuy** [k8up-io/k8up#671] [Q&A] [answered]
This project is very appealing to me. However I don't see an example or documentation on how to use the rest backend and what it actually does.

If I get thrown something so I can make it work for me I would definitely open a PR with improvements but I do need some breadcrumbs to follow so I can get started.

**Comment by SchoolGuy** [comment, marked as answer]
Is by any chance just the [rest-server](https://github.com/restic/rest-server) meant with the backend choice `rest`? Because if yes then this would already clear things up a lot for me.

**Reply by Kidswiss** [reply]
Hi

Yes rest is the restic rest-server.

**Comment by SchoolGuy** [comment]
So my tests at the moment are failing me. I was able to deploy k8up successfully and I annotated all my pvcs with `k8up.io/backup: "false"` so they are not backed-up per-default. I then used your example schedule to test it. Sadly I got an error when the job was executed.

I will try again on the weekend to try and pursue this adventure. Once I got this nut cracked I will try to open a PR for the docs as announced earlier with the knowledge I collected.

**Comment by SchoolGuy** [comment]
Okay dear diary... I continued my adventure. I got a little progress, however now my DNS is not resolving correctly and I forgot to set the right Prometheus URLs. Thus it seems the mist is lifted and I can continue to explore the depths of this application.

Okay serious now: I got huge progress as now I was able to contact the rest-server. The password for the repository was read from the secret and I finally realized that the CRDs are namespaces, so I need to add a schedule, secret and thus repo for every application I host (which is very very nice for me).

Sadly I got an additional error that one of the source files could not be read. I will try to investigate that as soon as my DNS problems are solved (they of course have nothing to do with k8up).

**Comment by SchoolGuy** [comment]
Okay last update for today: I got the first backup working with my rest-server. DNS issues are cleared but Prometheus is giving a 404 due to a misconfiguration from my side. I need to dig how to fix those and then I can start writing my findings down.
